src/Grpc.Net.Client/Internal/GrpcProtocolConstants.cs

```diff
@@ -83,6 +84,20 @@ internal static string GetMessageAcceptEncoding(Dictionary<string, ICompressionP
 #endif
     }
 
+    public static IEnumerator<KeyValuePair<string, object>> GetDebugEnumerator(ChannelBase channel, IMethod method, object? request)
```

Comment:
I feel like we've had this conversation before, but I've forgotten the output.  Why a list of pairs rather than a dictionary?

Reply:
It's easier to implement `IEnumerable` (one method) on the state than a dictionary (lots of methods). I'll add a comment explaining the choice.